spec/nsqlookup.yml

```diff
@@ -0,0 +1,448 @@
+  # /topics/{topic}/nodes/{node}:
+  #   delete:
+  /topic/tombstone:
+    # TODO: Consider 'delete' vs. 'post'
+    post:
+      operationId: doTombstoneTopicProducer
+      summary: Tombstones a specific producer of an existing topic.
+      description: |
+        Handles scenario of deleting a topic from a subset of nodes due to the additional complexities. Because of the way consumers query nsqlookupd and connect to all producers you enter into race conditions with attempting to remove the information from the cluster and consumers discovering that node and reconnecting (thus pushing updates that the topic is still produced on that node). The solution in these cases is to use “tombstones”. A tombstone in nsqlookupd context is producer specific and lasts for a configurable --tombstone-lifetime time. During that window the producer will not be listed in /lookup queries, allowing the node to delete the topic, propagate that information to nsqlookupd (which then removes the tombstoned producer), and prevent any consumer from re-discovering that node.
+      tags:
+        - nsqlookupd
+      parameters:
+        - name: topic
+          in: query
+          description: the name of topic to delete
+          type: string
+          required: true
+        - name: node
+          in: query
+          description: |
+            the producer (nsqd) to tombstone (identified by <broadcast_address>:<http_port>)
+          type: string
+          required: true
+      responses:
+        # TODO: Consider 201 or 204
+        200:
+          description: Success
+        400:
+          description: Bad request
+          schema:
+            type: object
+            properties:
+              message:
+                description: Specific error condition
+                type: string
+                default: INVALID_REQUEST
+                enum:
+                  - INVALID_REQUEST
+                  - MISSING_ARG_TOPIC
+                  - MISSING_ARG_NODE
+
+  # TODO: Alternate path based on REST
+  # /topics/{topic}/channels/{channel}:
+  /channel/create:
+    post:
+      operationId: doCreateChannel
+      summary: Creates a channel
+      description: |
+        Creates a channel and the topic if it does not already exist
+      tags:
+        - nsqlookupd
+      parameters:
+        - name: topic
+          in: query
+          description: the name of topic to create channel under
+          type: string
+          pattern: '^[\.a-zA-Z0-9_-]+(#ephemeral)?$'
+          minLength: 1
+          maxLength: 64
+          required: true
+        - name: channel
+          in: query
+          description: the name of channel to create
+          type: string
+          pattern: '^[\.a-zA-Z0-9_-]+(#ephemeral)?$'
+          minLength: 1
+          maxLength: 64
+          required: true
+      responses:
+        # TODO: Consider 201 or 204
+        200:
+          description: Success
+        # TODO: Consider CHANNEL_ALREADY_EXISTS

```

Comment:
I'd like to suggest considering 409 (Conflict) for CHANNEL_ALREADY_EXISTS as well.